src/Configuration/Api.php

```diff
@@ -153,18 +154,30 @@ private function createBaseUrl(array $api, array $configAttributes) : string
 
         if (filter_var($api['baseUrl'], FILTER_VALIDATE_URL)) {
             return $api['baseUrl'];
-        } elseif (is_string($api['baseUrl'])) {
+        }
+
+        if (is_string($api['baseUrl'])) {
             // For backwards compatibility
             try {
                 $fn = \Keboola\Utils\jsonDecode($api['baseUrl']);
                 $this->logger->warning("Passing json-encoded baseUrl is deprecated.");
             } catch (JsonDecodeException $e) {
                 throw new UserException("The 'baseUrl' attribute in API configuration is not a valid URL");
             }
-            return UserFunction::build([$fn], ['attr' => $configAttributes])[0];
+            $baseUrl = UserFunction::build([$fn], ['attr' => $configAttributes])[0];
         } else {
-            return UserFunction::build([$api['baseUrl']], ['attr' => $configAttributes])[0];
+            $baseUrl = UserFunction::build([$api['baseUrl']], ['attr' => $configAttributes])[0];
         }
+
+        if (!filter_var($baseUrl, FILTER_VALIDATE_URL)) {
+            throw new UserException(sprintf(
+                'The "baseUrl" attribute in API configuration (%s) resulted in an invalid URL (%s)',
+                json_encode($api['baseUrl']),
```

Comment:
json_encode($api['baseUrl']), nedaval bych to sem, muze to byt dost veliky, jestli jo, tak aspon otocit tu hlasku, at $baseUrl je ne zacatku, protoze to je to podstatnejsi

Reply:
OK. Fixed.